### ISSUE TITLE
Redefine emission of mouse-enter- and leave-events

### ISSUE THREAD

**Issue by neithernut** [waysome/waysome#424]
Make them modable via transactions/events

**Comment by TheNeikos** [comment]
How would you do that? As in, make a dedicated function that is called on the mouse?

I am not sure it's a good idea to make it interfaceable for the _mouse_. I am all for the keyboard though.

**Comment by neithernut** [comment]
Well, in `ws_cursor_set_position`, we currently determine whether the surface below a cursor changed and call `wl_pointer_send_enter()` and `wl_pointer_send_leave()` based on that data. I suggest basing these events on a transaction's return value, making emission the default (if, e.g. no such transaction exists).
Invocation can easily be done by calling `ws_action_manager_process()`, passing a prepared event. If a transaction exists, we'll get a value reply, if not, we'll get `NULL`.
Whether we require the transaction to emit the pointer and leave events on it's own or whether we delgate this to the caller by returning e.g. a bool is up to debate.

**Comment by TheNeikos** [comment]
I see, so what you mean is so that a plugin is able to actively decide if that window is 'allowed' to have the focus of a given window?

**Comment by neithernut** [comment]
Yes, e.g. if you just hover over another window but you don't want to have the current surface to loose focus _unless_ you click on the other one.

**Comment by matthiasbeyer** [comment]
@neithernut made a good point here.
